### ISSUE TITLE
Suggestion - Mechanic / Tow Job Improvements - Roadside Assistance 

### ISSUE THREAD

**Issue by CloudTheWolf** [InZidiuZ/op-framework-issue-tracker#2367]
**Assurance:**
Have you read through the rules from the `README.md` file in the root folder of this GitHub repository?
Yes
**Summary:**
Write a summary of what your suggestion is about.

Add a way for Mechanics/Tow Drivers to be able to do basic repairs at the roadside.

Eg Player has a crash and loses a tire or breaks the engine. they call a Roadside Assistance (Aka a Tow) who can review the vehicle which will tell them if they can get it working again of if it's too damaged and needs to be taken to a shop to repair.

If it can be repaired it does not fully repair the vehicle and can only be done to Engines (something like it repairs it up to 60%) and wheels/tires.

Additionally this could have Tires be items crafted by mechanics (Instead of how it currently works) that the can store in their truck to do roadside repairs.

This could also allow for more advanced features such as:

 - Window Repair Kit to fix Cracked Windows
 - Puncture Repair Kits (and Punctures, tyre pressure etc)
 - More advanced vehicle/engine damage.

This could either be with a specific vehicle (eg, See Below), or from any vehicle that you can use '/tow' from

Vehicle Example:
![Image](https://img.gta5-mods.com/q95/images/roadside-assistance-v-service-vehicle-f-350/d3be38-1.jpg)

**Reason:**
Why would your suggestion be a good change to the framework?

This would bring a more immersive experience to Mechanic RP

**Comment by MetaHatter** [comment]
I agree with making tires and windows into inventory items that people can use. 
Similar to the repair kits, but with guaranteed repair on use. (I would keep doors as an in-shop item only)

My thoughts:
- Still crafted by mechanics.  
- 10 weight each.
- Tires can be used by anyone 
- Windows require the mechanic (tow) role to use

**Comment by scufflord** [comment]
Having mechanics be able to go do mobile window repairs at car shows would be amazing. We just had a big car show event on international and there were quite a few cars missing a window or two here or there. This would be a perfect use case for it and allow collaboration between event organizers and a shop to offer window repairs on site.

**Comment by CloudTheWolf** [comment]
So, going back to this I had a few more ideas.

For Tow Driver: 
 - Can repair Doors, Windows and Tyres on the road 

For Intern Mechanic:
 - Can Craft Doors, Windows and Tyres to sell to Tow Drivers

For Mechanic:
 - Additionally can fix oil/fuel leaks on road side

For Lead Mechanic:
 - 10% faster at repairing body

For Expert Mechanic:
 - 10% faster at repairing engine

For Veteran Mechanic:
 - 10% Faster at inspecting 
 - 10% Faster at crafting

**Comment by coalaura** [comment]
An update on this. With the latest update you are able to remove entire wheels from vehicles using a drill. With the next update you will then receive a spare wheel item (you also get this item if you craft a wheel as a mechanic and then drop it using DEL). You can use the spare wheel to repair broken wheels which also repairs the tire. So theoretically you can craft spare wheels and actually change someones wheel no matter where

**Comment by coalaura** [comment]
Alright, this is basically done now. I've reworked the mechanic workbenches and the repair system entirely. The old script is now only for repairing the body and/or engine of a vehicle. All workbenches are now crafting tables instead allowing you to craft windows, wheels, doors and advanced repairkits as usual. However you get them as an item instead. Once you use those items you can then repair those vehicle parts anywhere
